credentialproviderpackage/internal/test/files.go

```diff
@@ -0,0 +1,137 @@
+package test
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+	"testing"
+
+	"credential-provider/pkg/filewriter"
+)
+
+var UpdateGoldenFiles = flag.Bool("update", false, "update golden files")
+
+func AssertFilesEquals(t *testing.T, gotPath, wantPath string) {
+	t.Helper()
+	gotFile := ReadFile(t, gotPath)
+	processUpdate(t, wantPath, gotFile)
+	wantFile := ReadFile(t, wantPath)
+
+	if gotFile != wantFile {
+		cmd := exec.Command("diff", wantPath, gotPath)
```

Comment:
Do we really need to diff? This adds a dependency to an OS binary which could be avoided by simply running a hash.